core/src/main/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpDocServicePlugin.java

```diff
@@ -120,14 +121,25 @@
     }
 
     @Override
-    public ServiceSpecification generateSpecification(Set<ServiceConfig> serviceConfigs) {
+    public ServiceSpecification generateSpecification(Set<ServiceConfig> serviceConfigs,
+                                                      BiPredicate<String, String> includeMethodPredicate,
+                                                      BiPredicate<String, String> excludeMethodPredicate) {
```

Comment:
Would it be possible to pass a single `BiPridicate` so that a plugin implementation does not have to repeat logical expressions?

Reply:
Just curious. The predicate checks whether the name matches exactly, but what if we use a pattern matching? I guess that some user might want to include/exclude the classes in some package or the methods of the name of a pattern. Would it be an overkill to use a pattern matching for this? 🤔

Reply:
Yeah, we can provide factory methods in `DocServiceFilter` which take in stings or patterns and which we can use it to do pattern matching. For example:
```java

static DocServiceFilter methodPattern(String methodPattern) {
    return (pluginName, serviceName, methodName) -> {
         // do the pattern matching here with methodName.
	}
}
```
,or the user can build his or her own `DocServiceFilter`. :)

Reply:
Never mind. Sorry, I mis-thought it. Need to get coffee... 😅